modules/aws_account_permissions/main.tf

```
@@ -0,0 +1,208 @@
+  role = aws_iam_role.illumio_cloud_integration_role.id
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow"
+        Action   = [
+          "ec2:AuthorizeSecurityGroupIngress",
+          "ec2:RevokeSecurityGroupIngress",
+          "ec2:UpdateSecurityGroupRuleDescriptionsIngress",
+          "ec2:AuthorizeSecurityGroupEgress",
+          "ec2:RevokeSecurityGroupEgress",
+          "ec2:UpdateSecurityGroupRuleDescriptionsEgress",
+          "ec2:ModifySecurityGroupRules",
+          "ec2:DescribeTags",
+          "ec2:CreateTags",
+          "ec2:DeleteTags",
+          "ec2:DescribeNetworkAcls",
+          "ec2:CreateNetworkAclEntry",
+          "ec2:ReplaceNetworkAclEntry",
+          "ec2:DeleteNetworkAclEntry"
+        ],
+        Resource = [
+          "arn:aws:ec2:*:*:security-group-rule/*",
+          "arn:aws:ec2:*:*:security-group/*",
+          "arn:aws:ec2:*:*:network-acl/*"
+        ]
+      }
+    ]
+  })
+}
+
+locals {
+  # Extract just the bucket ARN (without paths) for ListBucket actions
+  flow_logs_bucket_list = [
+    for bucket_arn in var.flow_logs_bucket_arns : regex("^arn:aws:s3:::[^/]+", bucket_arn)
+  ]
+
+  # Use the full ARN with path for GetObject actions
+  flow_logs_bucket_read = [
+    for bucket_arn in var.flow_logs_bucket_arns : "${bucket_arn}*"
+  ]
+}
+
+resource "aws_iam_role_policy" "IllumioCloudBucketListPolicy" {
+  count = length(local.flow_logs_bucket_list) > 0 ? 1 : 0
+  name  = "IllumioCloudBucketListPolicy"
+  role  = aws_iam_role.illumio_cloud_integration_role.id
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [{
+      Effect = "Allow"
+      Sid    = "IllumioBucketListAccess"
+      Action = [
+        "s3:ListBucket",
+        "s3:ListBucketVersion",
+        "s3:GetBucketLocation"
+      ]
+      Resource = local.flow_logs_bucket_list
+    }]
+  })
+}
+
+resource "aws_iam_role_policy" "IllumioCloudBucketReadPolicy" {
+  count = length(local.flow_logs_bucket_read) > 0 ? 1 : 0
+  name  = "IllumioCloudBucketReadPolicy"
+  role  = aws_iam_role.illumio_cloud_integration_role.id
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [{
+      Effect = "Allow"
+      Sid    = "IllumioBucketReadAccess"
+      Action = ["s3:GetObject"]
+      Resource = local.flow_logs_bucket_read
+    }]
+  })
+}
+
+resource "aws_iam_role" "illumio_cloud_integration_role" {
+  name = var.iam_role_name
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect    = "Allow"
+        Principal = {
+          AWS = "arn:${data.aws_partition.current.partition}:iam::137706418270:root"
+        }
+        Action    = "sts:AssumeRole"
+        Condition = {
+          StringEquals = {
+            "sts:ExternalId" = random_uuid.role_secret.result
+          }
+        }
+      }
+    ]
+  })
+
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/SecurityAudit"
+  ]
+}
```

Define the `illumio-cloudsecure_aws_account` resource directly in the module, instead of requiring users to do it themselves.
```suggestion
}

// Onboards this AWS account with CloudSecure.
resource "illumio-cloudsecure_aws_account" "account" {
  account_id       = data.aws_caller_identity.current.account_id
  mode             = var.mode
  name             = var.name
  organization_id  = data.aws_organizations_organization.current.id
  role_arn         = aws_iam_role.illumio_cloud_integration_role.arn
  role_external_id = random_uuid.role_secret.result
}
```